spec/system/admin/products_v3/products_spec.rb

```diff
@@ -450,4 +625,12 @@ def search_by_category(category)
   def row_containing_name(value)
     "tr:has(input[aria-label=Name][value='#{value}'])"
   end
+
+  # Wait for an element with the given CSS selector and class to be present
+  def wait_for_class(selector, class_name)
+    max_wait_time = Capybara.default_max_wait_time
+    Timeout.timeout(max_wait_time) do
+      sleep(0.1) until page.has_css?(selector, class: class_name, visible: false)
+    end
+  end
```

Comment:
This shouldn't be necessary because `has_css?` waits already. It *should* be the same as:
```suggestion
  # Wait for an element with the given CSS selector and class to be present
  def wait_for_class(selector, class_name)
    page.has_css?(selector, class: class_name, visible: false)
  end
```

But I guess that it didn't work? I wonder why...

Reply:
- It works when used with visible `visible: false` as suggested. 
- `page.has_css?(selector, class: class_name, visible: false)` returns a boolean and I thought it would make more sense to use it as a boolean. :sweat_smile: i.e. if the loader is still not hidden, then make sure to wait until it's hidden.
- If it is already hidden as the result of the wait in `has_css?` then the implemented `sleep(0.1)` won't be triggered.

If we still wanna commit the suggestion, then please let me know. No worries, we can. :)

Reply:
This PR is ready for testing already. I don't want to delay it. We can optimise specs another time.